### PR TITLE
Rotate bits using intrinsics

### LINKED ISSUE #2367: `<bit>`: Implement `rotl()` and `rotr()` with intrinsics
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/bit#L92-L118

`intrin.h` declares:

```cpp
__MACHINE(unsigned char __cdecl _rotl8(unsigned char _Value, unsigned char _Shift))
__MACHINE(unsigned short __cdecl _rotl16(unsigned short _Value, unsigned char _Shift))
__MACHINE(unsigned int __cdecl _rotl(_In_ unsigned int _Value, _In_ int _Shift))
__MACHINE(unsigned __int64 __cdecl _rotl64(_In_ unsigned __int64 _Value, _In_ int _Shift))

__MACHINE(unsigned char __cdecl _rotr8(unsigned char _Value, unsigned char _Shift))
__MACHINE(unsigned short __cdecl _rotr16(unsigned short _Value, unsigned char _Shift))
__MACHINE(unsigned int __cdecl _rotr(_In_ unsigned int _Value, _In_ int _Shift))
__MACHINE(unsigned __int64 __cdecl _rotr64(_In_ unsigned __int64 _Value, _In_ int _Shift))
```

* We'll need to move these to `intrin0.h` (physically, `intrin0.inl.h`) if we want throughput.
* We'll need to make sure this works for Clang.
* `<bit>` is C++20, so we don't need to worry about `/clr:pure` (for sure), CUDA (for sure), or the Intel C++ Compiler (I think).

Originally reported as DevCom-1590917 and VSO-1440998 / AB#1440998 .

### PR DESCRIPTION
**Pull request by AlexGuteniev**
Resolves #2367

### CODE REVIEW

**Comment by StephanTLavavej** [comment]
Note: This contains a TRANSITION comment because `cstdlib` provides only 2 of the 4 intrinsics. We could file a followup issue to use the remaining intrinsics after updating `intrin0.h`.

**Comment by AlexGuteniev** [comment]
Btw I confirm that clang-cl can use `_rotl16` and `_rotl8` and it doesn't mind moving the intrinsic from `<intrin.h>` elsewhere.

**Review by miscco — APPROVED** [review]

File: stl/inc/bit [hunk 1]
```diff
@@ -106,7 +116,17 @@ _NODISCARD constexpr _Ty rotl(const _Ty _Val, const int _Rotation) noexcept {
 template <class _Ty, enable_if_t<_Is_standard_unsigned_integer<_Ty>, int> _Enabled>
```
Comment (barcharcraz) [inline stl/inc/bit:116]: This enable if is using a nonstandard form, this is preexisting though

**Review by barcharcraz — APPROVED** [review]

**Review by cbezault — APPROVED** [review]

File: stl/inc/bit (hunk 1 above)
Comment (AlexGuteniev) [reply stl/inc/bit:116]: This is written this way in #310. And it is correct: there's a forward declaration with `= 0`, and the default can't be repeated.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for increasing the STL's angular momentum! When things spin, science happens. :gear: :milky_way: :rocket:

**Pull request merged**